### PR TITLE
Update most node and rails deps

### LINKED ISSUE #438: Some of the packages in node and some gems are very outdated - might require upgrade
I would say we take the upgradation process step by step in the node side. We could start with updating the `dev-dependencies` in package.json and then move to other ones if required. 

Similarly in the rails side, we could start updating the gems from `development` group first.

### Node Updations
image

### Rails Updations
```ruby
Outdated gems included in the bundle:
  * autoprefixer-rails (newest 10.0.2.0, installed 9.7.6, requested = 9.7.6) in group "default"
  * bootsnap (newest 1.5.1, installed 1.4.8) in group "default"
  * erubi (newest 1.10.0, installed 1.9.0)
  * font-awesome-sass (newest 5.15.1, installed 4.3.2.1, requested ~> 4.3.0) in group "default"
  * listen (newest 3.3.1, installed 3.2.1, requested ~> 3.2) in group "development"
  * mini_portile2 (newest 2.5.0, installed 2.4.0)
  * parallel (newest 1.20.0, installed 1.19.2)
  * parser (newest 2.7.2.0, installed 2.7.1.5)
  * puma (newest 5.0.4, installed 4.3.6, requested ~> 4.1) in group "default"
  * rubocop (newest 1.3.1, installed 0.82.0, requested ~> 0.82.0) in group "development"
  * rubocop-rails (newest 2.8.1, installed 2.6.0) in group "development"
  * simplecov (newest 0.19.1, installed 0.19.0) in group "test"
  * sprockets (newest 4.0.2, installed 3.7.2, requested ~> 3.7.2) in group "default"
  * tzinfo (newest 2.0.3, installed 1.2.7)
  * web-console (newest 4.1.0, installed 3.3.1, requested ~> 3.3.0) in group "development"
  * webpacker (newest 5.2.1, installed 4.3.0, requested ~> 4.0) in group "default"
  * zeitwerk (newest 2.4.1, installed 2.4.0)
```

@neerajdotname @vinay0x 
I can work on the PR for this. Can start with updating just the `dev-dependencies` in node side. Let me know your thoughts on this.

### PR DESCRIPTION
**Pull request by yedhink**
Fixes #438 

This upgrades overall outdated pkgs.

Some pkgs are intentionally left out of upgrade to latest tag, like tailwindcss and @rails/webpacker, cause the `postcss` loader is causing errors with rails webpacker. We can take these up in a separate issue when the upstream errors are fixed.

@vinay0x _a
Let me know if we need to make any changes.

Let's merge this only after other open `wheel` PR's that I made, are merged, so that we could test the functionalities once more after rebase. 

Also we should verify it in heroku, like Neeraj suggested. Let's do it together once other PR's are merged.

### CODE REVIEW

**Comment by vinay0x** [comment]
> Also we should verify it in heroku, like Neeraj suggested. Let's do it together once other PR's are merged.

Please verify now. No need to wait for the other PRs. It seems like deployment failed too. Can you reinstall packages in local and see if everything is working fine?

**Comment by yedhink** [comment]
@vinay0x _a

The problems with assets pipeline is now fixed. Deployment is success.

I verified the deployed version. Able to login/logout, add/delete notes, change password etc.

Please do a second verification from your side, of the deployed version. If it's ok, then I think it's safe to merge this.

File: app/views/layouts/application.html.slim [hunk 1]
```diff
@@ -6,12 +6,12 @@ html lang="en"
     meta content="width=device-width, initial-scale=1.0" name="viewport"
     title= content_for?(:title) ? yield(:title) : "Wheel"
     = csrf_meta_tags
+    = stylesheet_link_tag "application", media: "all"
+    = stylesheet_pack_tag "application"
```
Comment (vinay0x) [inline app/views/layouts/application.html.slim:10]: @yedhink _a Are both the tags necessary?

**Comment by yedhink** [comment]
@vinay0x _a

Have removed the `link_tag`. Also added the node-versioning files.  The reason from having `.nvmrc` too is to allow those who use nvm to auto switch to nvm version specified for that project when changing to that directory.

**Pull request merged**